src/main/java/net/dv8tion/jda/internal/utils/cache/MemberCacheViewImpl.java

```diff
@@ -93,10 +94,13 @@ public List<Member> getElementsWithRoles(@Nonnull Collection<Role> roles)
         Checks.noneNull(roles, "Roles");
         if (isEmpty())
             return Collections.emptyList();
+        List<Role> rolesWithoutPublicRole = roles.stream().filter(role -> !role.isPublicRole()).collect(Collectors.toList());
+        if (rolesWithoutPublicRole.isEmpty())
+            return asList();
         List<Member> members = new ArrayList<>();
```

Comment:
Looks much better now, thanks.

One nitty thing: I would perfer some new lines in between for a better readability.

```java
if (isEmpty())
    return Collections.emptyList()

List<Role> rolesWithoutPublicRole = roles.stream().filter(role -> !role.isPublicRole()).collect(Collectors.toList());
if (rolesWithoutPublicRole.isEmpty())
    return asList();

List<Member> members = new ArrayList<>();
```

But perhaps that's just me.

Reply:
agree